### ISSUE TITLE
Replace moment.js with date-fns 

### ISSUE THREAD

**Issue by PeterBreen** [DemocracyLab/CivicTechExchange#973]
Short version: Replace moment.js with date-fns, remove moment from our package.json, maintain all existing datetime functions.

Longer version:

Moment.js is quite large (the two biggest pieces of our bundle, per [this analysis of package.json are moment](https://bundlephobia.com/scan-results?packages=@tryghost/content-api@1.11.0,bluebird@3.5.2,django-react-csrftoken@1.0.4,flow-bin@0.75.0,flux@4.0.0,form-serialize@0.7.2,immutable@3.8.2,lodash@4.17.21,moment@2.22.2,moment-timezone@0.5.21,path@0.12.7,react@16.11.0,react-bootstrap@1.0.1,react-datepicker@2.14.1,react-dom@16.11.0,react-file-drop@0.2.8,react-google-recaptcha@1.1.0,react-helmet@5.2.0,react-image-crop@8.3.0,react-markdown@7.0.1,react-moment@0.7.9,react-select@2.4.2,react-sortablejs@6.0.0,repeating@2.0.0,sortablejs@1.13.0,validator@13.7.0&sortMode=size)) and the maintainers themselves [suggest moving to an alternative](https://momentjs.com/docs/#/-project-status/).

We use moment in [quite a few places](https://github.com/search?q=repo%3ADemocracyLab%2FCivicTechExchange%20moment&type=code) and any replacement must be able to handle all of them. The reason I specify date-fns is it's already a dependency of [react-datepicker](https://www.npmjs.com/package/react-datepicker), which we also use and want to keep at this time. 

To sum up, in order for this issue to be resolved,
 - add date-fns to our package.json (react-datepicker uses 2.30.0 - let's load this only once, so use the same version)
 - update all our datetime functions which use moment to use date-fns or vanilla js
 - remove moment, moment-timezone, and react-moment from package.json
 - enjoy new, smaller bundle size

**Comment by PeterBreen** [comment]
I'd also suggest making further use of our datetime utility component, seen here: https://github.com/DemocracyLab/CivicTechExchange/blob/master/common/components/utils/datetime.js

If we can collect all our datetime needs and create reusable functions here and use them in the actual page components, future upgrades or changes to our datetime library will be editing one file in one place, not a dozen files in a dozen places.

**Comment by BQingFan** [comment]
Could I try to solve this issue? I just finished environment set up.

**Comment by PeterBreen** [comment]
> Could I try to solve this issue? I just finished environment set up.

Absolutely. I'll assign you, feel free to reach out if you're having trouble. There's a lot of components we use moment for, but hopefully it'll make sense how they operate.

**Comment by PeterBreen** [comment]
> I was wondering if I could try to work on this issue?

Sure, please do. I checked with Marlon and we think it's open for the taking so it's all yours.

**Comment by Kive1ru** [comment]
Is this issue solved? Can I try to work on this issue if not solved?

**Comment by PeterBreen** [comment]
PR #1035 solves this issue, pending a couple bugfixes. I believe Kyle's still working on it.
